### ISSUE TITLE
Implicit declarations during compilation

### ISSUE THREAD

**Issue by ryanaddams** [hoedown/hoedown#158]
Minor cosmetics, but since I hate seeing any warnings during compilation:

`
src/buffer.c: In function ‘hoedown_buffer_printf’:
src/buffer.c:249:2: warning: implicit declaration of function ‘vsnprintf’ [-Wimplicit-function-declaration]
  n = vsnprintf((char *)buf->data + buf->size, buf->asize - buf->size, fmt, ap);
`

`
src/html_smartypants.c: In function ‘smartypants_quotes’:
src/html_smartypants.c:102:2: warning: implicit declaration of function ‘snprintf’ [-Wimplicit-function-declaration]
  snprintf(ent, sizeof(ent), "&%c%cquo;", (*is_open) ? 'r' : 'l', quote);
`

Both can be solved by adding `#include <stdarg.h>` to `src/html_smartypants.c` and `src/buffer.c` file headers.

**Comment by jmendeth** [comment]
Would they? For glibc we need C99 or feature flags set...

**Comment by jmendeth** [comment]
For me, including `stdarg.h` does not remove the warnings. I have gcc version 4.8.2 (Ubuntu 4.8.2-19ubuntu1). I have glibc version 2.19-0ubuntu6.6. Working on `master`.

**Comment by uranusjr** [comment]
Just to make sure, does v4 continue to stick to ANSI C like previous versions?

**Comment by jmendeth** [comment]
Yes, it's one of the objectives for the final release. For now though, I'm sticking to C99 and CMake.

**Comment by ryanaddams** [comment]
Oh, it's because of the -ansi.

**Comment by jmendeth** [comment]
Yep. We used to leave the default GNU quirky mode, but @devinus finally made it strictly C89. I don't know what to do with that function though...

**Comment by jmendeth** [comment]
The right way to avoid the warnings would be to set a feature flag. Any of these should do:

~~~ c
#define _BSD_SOURCE
#define _XOPEN_SOURCE 500
#define _ISOC99_SOURCE /* not what we want */
#define _POSIX_C_SOURCE 200112L
~~~
